### PR TITLE
Don't filter own toots (fixes #8289)

### LINKED ISSUE #8289: [Feature] Don't filter my own posts (or option to show own posts?)
Say if I'm filtering the word "Trump", if I post saying "I'm filtering the word Trump, just so you know" then that post isn't visible to me.

I'd like to be able to see my own posts (which are clearly acceptable and not triggering to me, because I wrote them). If that's not a universal want, then I'd like to have the option. Maybe in amongst these settings?:

screen shot 2018-08-19 at 09 52 08

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by ClearlyClaire**
fixes #8289

### CODE REVIEW

**Review by nightpool — COMMENTED** [review]
does this also fix "drop instead of hide" filters or are those not broken?

File: app/javascript/mastodon/selectors/index.js [hunk 1]
```diff
@@ -83,7 +84,7 @@ export const makeGetStatus = () => {
         statusReblog = null;
       }
 
-      const regex    = regexFromFilters(filters);
+      const regex    = (accountReblog || acountBase).get('id') !== me && regexFromFilters(filters);
```
Comment (Gargron) [inline app/javascript/mastodon/selectors/index.js]: acount -> account

**Comment by ThibG** [comment]
@nightpool they aren't broken, own toots are never filtered in the backend afaict
https://github.com/tootsuite/mastodon/blob/20fefdb7145b48449e9d6a824a9846dadfd6ac2f/app/lib/feed_manager.rb#L153-L154

**Comment by nightpool** [comment]
sgtm!

**Review by Gargron — APPROVED** [review]

**Pull request merged**